src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -43,43 +43,31 @@ static void SetSchemaNameIfNotExist(char **schemaName, char *newSchemaName);
 
 /*
  * RelayEventExtendNames extends relation names in the given parse tree for
- * certain utility commands. The function more specifically extends table,
- * sequence, and index names in the parse tree by appending the given shardId;
- * thereby avoiding name collisions in the database among sharded tables. This
- * function has the side effect of extending relation names in the parse tree.
+ * certain utility commands. The function more specifically extends table and
+ * index names in the parse tree by appending the given shardId; thereby
+ * avoiding name collisions in the database among sharded tables. This function
+ * has the side effect of extending relation names in the parse tree.
  */
 void
 RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 {
 	/* we don't extend names in extension or schema commands */
 	NodeTag nodeType = nodeTag(parseTree);
-	if (nodeType == T_CreateExtensionStmt || nodeType == T_CreateSchemaStmt)
+	if (nodeType == T_CreateExtensionStmt || nodeType == T_CreateSchemaStmt ||
+		nodeType == T_CreateSeqStmt || nodeType == T_AlterSeqStmt)
 	{
 		return;
 	}
 
 	switch (nodeType)
 	{
-		case T_AlterSeqStmt:
-		{
-			AlterSeqStmt *alterSeqStmt = (AlterSeqStmt *) parseTree;
-			char **sequenceName = &(alterSeqStmt->sequence->relname);
-			char **sequenceSchemaName = &(alterSeqStmt->sequence->schemaname);
-
-			/* prefix with schema name if it is not added already */
-			SetSchemaNameIfNotExist(sequenceSchemaName, schemaName);
-
-			AppendShardIdToName(sequenceName, shardId);
-			break;
-		}
```

Comment:
The short of is that there isn't a correctness reason for sequence related statements to be here.

When we were writing this code, I went over all DDL statements Postgres supported and grouped them into buckets. Sequences and tables shared similar code paths, so I thought that if we extended table names, we should also extend sequence names. So the primary motivation was to maintain some form of symmetry between related DDL commands.